platforms/chibios/drivers/eeprom/eeprom_stm32_L0_L1.c

```diff
@@ -86,17 +86,35 @@ void eeprom_read_block(void *buf, const void *addr, size_t len) {
 }
 
 void eeprom_write_block(const void *buf, void *addr, size_t len) {
-    STM32_L0_L1_EEPROM_Unlock();
+    // use word-aligned write to overcome issues with writing null bytes
+    uint32_t start_addr = (uint32_t)addr;
+    uint32_t end_addr   = start_addr + len;
```

Comment:
A fully correct check (which also catches some really pathological cases like `addr` being out of range, or `start_addr + len` overflowing and wrapping around to a seemingly valid offset) may look like this (assuming that `STM32_ONBOARD_EEPROM_SIZE` is properly aligned):
```suggestion
    uint32_t start_addr = (uint32_t)addr;
    if (start_addr >= (STM32_ONBOARD_EEPROM_SIZE)) {
        return;
    }
    uint32_t max_len = (STM32_ONBOARD_EEPROM_SIZE) - start_addr;
    if (len > max_len) {
        len = max_len;
    }
    uint32_t end_addr = start_addr + len;
```